setup.md

```diff
@@ -269,8 +269,124 @@ TODO: Explain one of the following:
+
+**TODO: Add and link to section on configuring environment to detect local rocks**
+
+#### Manual installation
+
+See the [installation guide in the readline.lua docs](https://pjb.com.au/comp/lua/readline.html#installation)
+for a URL from which to obtain the source code of the most recent version.
+The source is also available, along with the author's other Lua modules, on
+[Gitlab](https://gitlab.com/peterbillam/pjb_lua).
+
+To manually compile the native bindings, you can use `gcc`. Here is an example
+on a typical Linux system:
+
+```bash
+$ gcc -O2 -fPIC -c C-readline.c -o C-readline.o
+$ gcc -shared -o C-readline.so C-readline.o -lreadline
+
+# to compile for a specific Lua version, you will need to include its location:
+$ gcc -O2 -fPIC -I/usr/include/luajit-2.0 -c C-readline.c -o C-readline.o
+$ gcc -shared -o C-readline.so C-readline.o -lreadline
+```
+
+Once the bindings are compiled, simply ensure readline.lua is available on your
+`package.path` and C-readline.so on `package.cpath`.
+
+### Configuring readline for an enhanced experience
+
+Readline itself has a number of configuration options, which can be set either
+via the readline.lua API in `fennelrc`, or in readline's own `~/.inputrc` config file.
+
+#### Enabling persistent history
+
+To configure the REPL to save the rolling history to file at the end of every
+session, add the following to your `fennelrc` with your desired filename:
+
+See the readline.lua documentation for information on its API, most notably
+other parameters that can be set via
+[rl.set_options](https://pjb.com.au/comp/lua/readline.html#set_options).
+
+```fennel
+; persist repl history
+(match package.loaded.readline
+  rl   (rl.set_options {:histfile  "~/.fennel_history" ; default:"" (don't save)
+                        :keeplines 1000}))             ; default:1000
+```
+
+#### Configuring readline in `~/.inputrc`
+
+[documentation on the readline init file](https://www.gnu.org/software/bash/manual/html_node/Readline-Init-File.html)
+for the full set of options and a sample inputrc.
+
+As of Fennel 0.4.0 and readline.lua 2.6, you can make use of a [conditional
+directive your `inputrc`](https://www.gnu.org/software/bash/manual/html_node/Conditional-Init-Constructs.html#Conditional-Init-Constructs)
+for fennel-only configuration options. 
+
+The following example adds these behaviors:
+- Blink on a matching parenthesis when entering `)`. Useful in a Lisp REPL, where
+the parens are plentiful!
+- Enable bracketed paste mode for more reliable pasting from clipboard
+- When tab-completing on a term with more than one possible match, display all
+candidates immediately instead of ringing the bell + requiring a second `<tab>`
+
+```inputrc
+$if fennel
+  set enable-bracketed-paste on
+  set blink-matching-paren on
+  set show-all-if-ambiguous on
+$endif
+```
+
+**NOTE:** The conditional directive `$if fennel` used above to set fennel-only
```

Comment:
I think it's OK to assume the latest versions of each of these in the setup guide, since you won't get an older version if you follow the instructions in this file?

Reply:
True! And feel free to edit it at will, but I'd like to mention the requirements for `.inputrc` to work *somewhere* just it's written down somewhere people can find and we can link to in the event people take note of their `.inputrc` doing nothing.

This is the first time we documented the readline stuff, so it's possible very few people are currently using it, so I'm guessing there will be a lot of people who start using it. We might want to make mention of it in the changelog, just so people not using emacs know they can get some bells and whistles?